OpenRA.Mods.Common/Activities/Air/HeliAttack.cs

```diff
@@ -41,10 +51,10 @@ public override Activity Tick(Actor self)
 				return this;
 
 			// Fly towards the target
-			if (!target.IsInRange(self.CenterPosition, attack.GetMaximumRange()))
+			if (!target.IsInRange(self.CenterPosition, attackHeli.GetMaximumRange()))
```

Comment:
This leads to the helicopter not doing anything when the armament with the longest range is out of ammo. (As shown with the hind testcase).  `FlyAttack` uses the minimum range of all armament instead (which probably leads to planes not shooting as early as they could under circumstances). Ideally, `GetMaximumRange` and `FlyAttacks`'s `Tick()` should check for the weapon's validity against the target and the state of its ammo pool.

This behaviour isn't this PR's fault though. But maybe leave a comment?